lib/vintage_net_wizard/wifi_configuration.ex

```diff
@@ -3,77 +3,98 @@ defmodule VintageNetWizard.WiFiConfiguration do
   Module for working with various WiFi configurations
   """
 
-  @opaque t ::
-            VintageNetWizard.WiFiConfiguration.WPAPersonal.t()
-            | VintageNetWizard.WiFiConfiguration.NoSecurity.t()
-            | VintageNetWizard.WiFiConfiguration.PEAPEnterprise.t()
+  @type key_mgmt() :: :none | :wpa_psk | :wpa_eap
 
-  @type key_mgmt() :: :none | :wpa_psk
+  @type configuration() :: %{
+          :ssid => String.t(),
+          :key_mgmt => key_mgmt(),
+          :mode => :infrastructure,
+          :priority => non_neg_integer() | nil,
+          optional(:psk) => String.t(),
+          optional(:identity) => String.t(),
+          optional(:password) => String.t(),
+          optional(:eap) => String.t(),
+          optional(:phase2) => String.t()
+        }
 
-  alias VintageNetWizard.WiFiConfiguration.{
-    Params,
-    WPAPersonal,
-    NoSecurity,
-    PEAPEnterprise
-  }
+  alias VintageNetWizard.WiFiConfiguration.Params
 
   @doc """
-  A function on how to transform the map of params into a WiFiConfiguration
+  Make a `VintageNet` configuration from the params
   """
-  @callback from_params(map()) :: {:ok, t()} | {:error, Params.param_error()}
-
-  @doc """
-  A function on how to take a WiFiConfiguration and make it into
-  a useable `vintage_net` WiFi configuration. 
-  """
-  @callback to_vintage_net_configuration(t()) :: map()
+  @spec from_params(map()) :: {:ok, configuration()} | {:error, Params.param_error()}
```

Comment:
If this is a VintageNetWiFi configuration, then the type is `map()`. It feels weird for it not to be the whole configuration, though (the IPv4 and type stuff too).

```elixir
%{
    type: VintageNetWiFi,
    vintage_net_wifi: %{
      mode: :infrastructure,
      networks: [%{ssid: "my_network_ssid", key_mgmt: :wpa_psk, psk: "a_passphrase_or_psk"}]
    },
    ipv4: %{method: :dhcp}
  }
  ```

Reply:
I think my goal was to have it be the configuration maps that in is the `:network` field.

I am okay with switching to just being the whole vintage net configuration, but this function's primary goal is taking a JSON packet and turning into a configuration for the `:networks` field - maybe the goal should change?

We can return the full vintage net configuration from this function, however I think the caller will need to handle merging two different complete vintage net configurations - which my assumption will just be appending the lists together so it shouldn't be too much work.

I would suggest maybe that the backend server's `configuration` state be the whole vintage net configuration and once we handle the JSON packets we can append that to the `:networks` list. Currently, it just a map with the ssid as keys - which makes it easy to query information via the ssid.

Currently, when we apply the configuration we just take the list of configurations and drop them into the `:network` field:https://github.com/nerves-networking/vintage_net_wizard/blob/vintage-net-configs/lib/vintage_net_wizard/backend/default.ex#L150-L158